src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -262,6 +305,39 @@ public ZipFile(final File f, final String encoding, final boolean useUnicodeExtr
              f.getAbsolutePath(), encoding, useUnicodeExtraFields, true, ignoreLocalFileHeader);
     }
 
+    /**
+     * Opens the given path for reading, assuming the specified
+     * encoding for file names.
+     *
+     *
```

Comment:
Ditto

Reply:
This is a copy paste from the existing javadocs of methods from within the class, if you want I can correct them all for white-spaces.